service/frontend/workflow_handler.go

```diff
@@ -6022,23 +6022,34 @@ func (wh *WorkflowHandler) UpdateTaskQueueConfig(
 	if err != nil {
 		return nil, err
 	}
+
 	// Validation: prohibit setting rate limit on workflow task queues
 	if request.TaskQueueType == enumspb.TASK_QUEUE_TYPE_WORKFLOW {
 		return nil, serviceerror.NewInvalidArgument("Setting rate limit on workflow task queues is not allowed.")
 	}
```

Comment:
```suggestion
    queueRateLimit := request.GetUpdateQueueRateLimit()
    
	// Validation: prohibit setting queue rate limit on workflow task queues
	// Allow fairness-related settings (fairness key rate limits and fairness weight overrides) on workflow task queues
	if request.TaskQueueType == enumspb.TASK_QUEUE_TYPE_WORKFLOW {
		if queueRateLimit != nil {
			return nil, serviceerror.NewInvalidArgument("Setting queue rate limit on workflow task queues is not allowed.")
		}
	}
```
Currently, in the frontend routing for UpdateTaskQueueConfig, we block updates on workflow task queues because setting queue-level rate limits are not allowed for them. However, now that the fairness-related config changes have been merged, this restriction needs to be revised.
Since there are no restrictions on setting fairness key weight overrides or default fairness key rate limits for workflow task queues, we should refine the check. Specifically, requests that attempt to configure a queue rate limit on a workflow task queue should be rejected, but requests that update fairness-related configs should be allowed.

Reply:
Thank you; excellent catch!